### PR TITLE
BUG: added nbio websocket tls example

### PR DESCRIPTION
**Pull request by acgreek**
the client panics on the resp being nil

### CODE REVIEW

File: examples/websocket_tls/nbioclient/client.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	if err != nil {
+		fmt.Printf("nbio.Start failed: %v\n", err)
+		return
+	}
+
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: true,
+	}
+	for i := 0; i < 1; i++ {
+		u := url.URL{Scheme: "ws", Host: "localhost:8888", Path: "/ws"}
```
Comment (lesismal) [inline examples/websocket_tls/nbioclient/client.go:45]: Here should be:
```golang
u := url.URL{Scheme: "wss", Host: "localhost:8888", Path: "/wss"}
```

File: examples/websocket_tls/nbioclient/client.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+	u.OnClose(func(c *websocket.Conn, err error) {
+		fmt.Println("OnClose:", c.RemoteAddr().String(), err)
+	})
+
+	return u
+}
+
+func main() {
+	engine := nbhttp.NewEngine(nbhttp.Config{})
```
Comment (lesismal) [inline examples/websocket_tls/nbioclient/client.go:34]:
```golang
engine := nbhttp.NewEngineTLS(nbhttp.Config{})
```

or 
```golang
engine := nbhttp.NewEngine(nbhttp.Config{})
err := engine.Start()
if err != nil {
	fmt.Printf("nbio.Start failed: %v\n", err)
	return
}
engine.InitTLSBuffers()
```

File: examples/websocket_tls/nbioclient/client.go (hunk 1 above)
Comment (acgreek) [reply examples/websocket_tls/nbioclient/client.go:45]: ok, but same results

File: nbhttp/websocket/dialer.go [hunk 3]
```diff
@@ -155,6 +155,9 @@ func (d *Dialer) DialContext(ctx context.Context, urlStr string, requestHeader h
 		}
 
 		remoteCompressionEnabled := false
+		if resp == nil {
+			panic("response should not be nil")
+		}
```
Comment (lesismal) [inline nbhttp/websocket/dialer.go:160]: We should not panic except we got an error during the server init period.

File: examples/websocket_tls/nbioclient/client.go (hunk 2 above)
Comment (acgreek) [reply examples/websocket_tls/nbioclient/client.go]: still panics

Comment (lesismal) [reply examples/websocket_tls/nbioclient/client.go]: And here, it will be ok.

File: nbhttp/websocket/dialer.go (hunk 3 above)
Comment (acgreek) [reply nbhttp/websocket/dialer.go:160]: right, it should not get here. I just put it there for debugging. err == nil and resp == nil should not happen

File: examples/websocket_tls/nbioclient/client.go (hunk 2 above)
Comment (lesismal) [reply examples/websocket_tls/nbioclient/client.go]: > still panics

But I run it fine:
```golang
// client.go
package main

import (
	"context"
	"fmt"
	"log"
	"net/url"
	"os"
	"os/signal"
	"time"

	"github.com/lesismal/nbio/nbhttp"
	"github.com/lesismal/nbio/nbhttp/websocket"

	"github.com/lesismal/llib/std/crypto/tls"
)

func newUpgrader() *websocket.Upgrader {
	u := websocket.NewUpgrader()
	u.OnMessage(func(c *websocket.Conn, messageType websocket.MessageType, data []byte) {
		// echo
		c.WriteMessage(messageType, data)
		log.Println("onEcho:", string(data))
	})

	u.OnClose(func(c *websocket.Conn, err error) {
		fmt.Println("OnClose:", c.RemoteAddr().String(), err)
	})

	return u
}

func main() {
	engine := nbhttp.NewEngineTLS(nbhttp.Config{})
	err := engine.Start()
	if err != nil {
		fmt.Printf("nbio.Start failed: %v\n", err)
		return
	}

	tlsConfig := &tls.Config{
		InsecureSkipVerify: true,
	}
	for i := 0; i < 1; i++ {
		u := url.URL{Scheme: "wss", Host: "localhost:8888", Path: "/wss"}
		c, _, err := (&websocket.Dialer{
			TLSClientConfig: tlsConfig,
			Engine:          engine,
		}).Dial(u.String(), nil, newUpgrader())
		if err != nil {
			panic(fmt.Errorf("dial: %v", err))
		}
		c.WriteMessage(websocket.TextMessage, []byte("hello"))
	}

	interrupt := make(chan os.Signal, 1)
	signal.Notify(interrupt, os.Interrupt)
	<-interrupt
	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
	defer cancel()
	engine.Shutdown(ctx)
}
```

Comment (acgreek) [reply examples/websocket_tls/nbioclient/client.go]: with this, I get: `dial: local error: tls: unexpected message`

**Pull request merged**